core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -1558,7 +1559,7 @@ class ReplicaManager(val config: KafkaConfig,
     }
 
     val remoteFetchMaxWaitMs = config.remoteLogManagerConfig.remoteFetchMaxWaitMs().toLong
-    val remoteFetch = new DelayedRemoteFetch(remoteFetchTask, remoteFetchResult, remoteFetchInfo, remoteFetchMaxWaitMs,
+    val remoteFetch = new DelayedRemoteFetch(remoteFetchFuture, remoteFetchInfo, remoteFetchMaxWaitMs,
```

Comment:
Since we only need to pass in `remoteFetchTask` and `remoteFetchResult`, it seems that it's simpler to just pass them in separately as before, instead of wrapping them under `RemoteFetchFuture`.

Reply:
Right. In that case the main change would be cancelling the future without interrupting it.